server/src/main.rs

```diff
@@ -37,12 +37,24 @@ async fn main() -> Result<(), BoxedStdError> {
 
     let args = cli::parse();
 
-    let data = fs::read_to_string(args.path)
-        .unwrap_or_else(|error| panic!("Could not read the startup state.\nError: {error}"));
+    let data = fs::read_to_string(args.path).unwrap_or_else(|error| {
+        use std::io::ErrorKind as EK;
+        match error.kind() {
+            EK::NotFound => log::error!("Startup configuration file not found."),
+            EK::InvalidData => log::error!("Startup configuration file contains non UTF-8 data."),
+            e => log::error!("Error reading startup configuration file: {}.", e),
```

Comment:
Outputting the error as in the last case would actually already do the job without overcomplicating things.
Additionally it would allow to unify the handling and avoiding repetition.